### PR TITLE
LIVE-9378: add Sonarqube LLC integration

### PR DESCRIPTION
**Pull request by mohammed-ledger**
### 📝 Description

Integrate Ledger live common Sonarqube code scanning
### ❓ Context

- **Impacted projects**: `` 
- **Linked resource(s)**: `` 

### ✅ Checklist

- [ ] **Test coverage** 
- [ ] **Atomic delivery** 
- [ ] **No breaking changes** 

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

### CODE REVIEW

File: .github/workflows/build-sonarqube.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+name: "[Bot] Run sonarqube code scanning'"
+on:
+  push:
+    branches:
+      - develop
```
Comment (gre) [inline .github/workflows/build-sonarqube.yml:5]: were you able to test it yet?
if not, you can always add a temporary `- support/LIVE-9378-add-sonarqube-integration` after the `- develop`

Comment (mohammed-ledger) [reply .github/workflows/build-sonarqube.yml:5]: Not yet, good suggestion let me try it.

File: .github/workflows/build-sonarqube.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
+name: "[Bot] Run sonarqube code scanning'"
+run-name: "@Desktop • Test App triggered by ${{ github.event_name == 'workflow_dispatch' && inputs.login || github.actor }} ${{ format('on ref {0}', github.ref_name) }}"
+
+on:
+  push:
+    branches:
+      - develop
+      - support/LIVE-9378-add-sonarqube-integration
+  workflow_dispatch:
```
Comment (mbertin-ledger) [inline .github/workflows/build-sonarqube.yml:9]: Why we would run it manually ?

File: .github/workflows/build-sonarqube.yml [hunk 3]
```diff
@@ -0,0 +1,55 @@
+name: "[Bot] Run sonarqube code scanning'"
+run-name: "@Desktop • Test App triggered by ${{ github.event_name == 'workflow_dispatch' && inputs.login || github.actor }} ${{ format('on ref {0}', github.ref_name) }}"
+
+on:
+  push:
+    branches:
+      - develop
+      - support/LIVE-9378-add-sonarqube-integration
```
Comment (mbertin-ledger) [inline .github/workflows/build-sonarqube.yml:8]: [TODO] Remove it before merge :)

File: .github/workflows/build-sonarqube.yml (hunk 2 above)
Comment (mohammed-ledger) [reply .github/workflows/build-sonarqube.yml:9]: @mbertin-ledger nice catch, this is not needed. I copied the desktop testing workflow for the initial setup, the Sonarqube testing step will possibly be moved to test.yml flow after @gre's feedback so we won't run the app tests various times.

File: sonar-project.properties [hunk 4]
```diff
@@ -0,0 +1,6 @@
+sonar.projectKey=LedgerHQ_ledger-live_AYoXRskslck2EnGRKg2f
+sonar.sources=apps/ledger-live-desktop/src/
+sonar.exclusions=**/*.java,**/node_modules/,**/.turbo,**/.pnpm-*,**/temp,**/.eslintcache,**/tsconfig.tsbuildinfo,**/*.png,**/*.env
+sonar.coverage.exclusions=**/*.java
+sonar.tests=apps/ledger-live-desktop/tests/
```
Comment (aboissiere-ledger) [inline sonar-project.properties:5]: This is actually Playwright folder. It should only contain e2e tests. Unit tests are in the src folder.

File: sonar-project.properties [hunk 5]
```diff
@@ -0,0 +1,6 @@
...
+sonar.javascript.lcov.reportPaths=apps/ledger-live-desktop/coverage/lcov.info
```
Comment (aboissiere-ledger) [inline sonar-project.properties:6]: If you want to see if LLC coverage can be reported to, here's LLC report path: libs/ledger-live-common/coverage/lcov.info

**Review by valpinkman — APPROVED** [review]

**Comment by valpinkman** [comment]
Please do squash and merge or rewrite a bit the commit history as the integration is only a few files but we have 30 commits

**Pull request merged**